### ISSUE TITLE
Problem with reportIssue and include

### ISSUE THREAD

**Issue by samixchoumi** [Riverside-Software/sonar-openedge#267]
I've a file called test.p .
This file use some include.
I created a rule that reportIssue when `QUERY` is find.
Issues in test.p are good, but sometime i found issues on commented lines or at several locations where `QUERY` isn't here. And after looking in the proparse scan, i saw that include file was scanned but the line doesn't correspond as a PREPROCESS-test.p, just the include file is scanned alone and if `QUERY` is find on the include file of the test.p, the issue is reported on the test.p file.
So the issue showed in test.p, in a commented line is the issue found on one of the include of test.p
(i can see what include is scanned with this `Arrays.toString(node.getDirectChildrenArray())`).

So my question is, does this problem is from proparse or from the plugin ?

**Comment by gquerret** [comment]
Likely to come from the parser, but then again, a more detailed example would be useful to fix the issue.

**Comment by samixchoumi** [comment]
Here it's what i put on the log to see what is the file parsed : 
```
[sonar:sonar] |==========================================|
[sonar:sonar] | node type  := 192                        |
[sonar:sonar] | node text  := DEFINE                     |
[sonar:sonar] | node line  := 38                         | 
[sonar:sonar] | ---------------------------------------- |
[sonar:sonar] | node.directchildrens := [QUERY "QUERY" browser.std:38:19] |
[sonar:sonar] |==========================================|
```
So the file scanned here is the include browser.std but the main file is test.p
So the issue reported gave me this : 
![bug_proparse](https://cloud.githubusercontent.com/assets/22171928/26715387/db26e43c-4775-11e7-9a14-346d1de10f99.png)

If you need more information, tell me what you need and i will give you those information

**Comment by gquerret** [comment]
As said, that's probably coming from the preprocessor or from the parser, but without a test case I can reproduce, I can't fix the problem.

**Comment by samixchoumi** [comment]
For this test case we will have a file called "test.p".
We will add approximately 150 lines of commentary at the start, normaly it's the log of the procedure.
Some line after, defining some variable and call some include, we will create a procedure. 
This procedure could be like that :

```
PROCEDURE Test_Table:
 
    DEFINE INPUT  PARAMETER test1           AS CHARACTER    NO-UNDO.
    DEFINE INPUT  PARAMETER test2           AS CHARACTER    NO-UNDO.
    DEFINE INPUT  PARAMETER test3           AS INTEGER      NO-UNDO.
 
    IF AVAILABLE test1
    THEN DO :
        ASSIGN test2 = "The command " + test3 + " is being edited.":L .
    END.

    {testInclude.i02}
END PROCEDURE. /* Test_table */
```

Then the include should have at least 97 lines of code (commentary) and at the 98 line, you put a CREATE statement.
So, the rules is simple, when the code encounter a "CREATE" it will tell us that CREATE has been found. 
And then, at line 98 of the procedure test.p, i will have in my comentary an issue about the CREATE statement of the include file testInclude.i02 .

It worked for me in this test case and the same thing appear in my real project.

**Comment by gquerret** [comment]
Which version of the plugin do you use ?

**Comment by samixchoumi** [comment]
Release : sonar-openedge-plugin-2.0.0

**Comment by samixchoumi** [comment]
I allow myself to re-open this issue, i was doing a new scan with the release 2.0.1 and all depencies updated and that was fine, but after looking at some of my project, sometime i've seen the same error as i told you before. 
So, if the bug was corrected on the release 2.0.1, what should i do after updating the release and the dependencies ?

**Comment by gquerret** [comment]
Given your example test3.p, keyword is detected in the right file at the right line number. So what's wrong ?
I don't understand the latest comment on this issue.

Push your code to a GitHub repository if you want more help, I can't help with only fragments of code.
